### PR TITLE
Enforce file extensions in imports to comply with ESM standards

### LINKED ISSUE #454: Use proper ESM imports w/ extensions
### Feature description

Remove the `esm` and `extensionless` package usage in the repository, updating all imports throughout the project to use proper ESM imports with extensions, ideally marking the package as a proper module.

### How the feature is useful

Stop relying on janky workarounds to improper ESM usage.

### PR DESCRIPTION
**Pull request by dmz-v-x**
### **Type of Change:**  
Tool Source: JS

### **What issue does this relate to?**  
This PR resolves #454.

### **What should this PR do?**  
- Removed all `esm + extensionless` usages from the project.  
- Updated ESLint configuration to enforce file extensions in import statements.  
- Ensured all imports in the project now include proper file extensions.  
- Updated the project’s `package.json` to set `"type": "module"`.  
- Updated the year in the copyright header for all modified files.

### **What are the acceptance criteria?**  
- No remaining extensionless imports.  
- ESLint configuration correctly enforces file extensions.  
- Copyright headers are updated.

---

**Feedback:**  
Please review and let me know if any further adjustments are required. Thank you!

### CODE REVIEW

**Comment by MattIPv4** [comment]
👋 Just as an FYI for the future, you don't need to create a new PR to make changes to it, you should just push to your existing PR. cc #485

File: src/nginxconfig/i18n/ru/index.js
Comment (MattIPv4) [inline src/nginxconfig/i18n/ru/index.js:1]: Copyright header not updated

File: src/nginxconfig/i18n/ru/templates/domain_sections/logging.js
Comment (MattIPv4) [inline src/nginxconfig/i18n/ru/templates/domain_sections/logging.js:1]: Copyright header not updated

File: src/nginxconfig/util/analytics.js
Comment (MattIPv4) [inline src/nginxconfig/util/analytics.js:1]: Copyright header not updated

File: src/nginxconfig/util/angular_backwards_compatibility.js
Comment (MattIPv4) [inline src/nginxconfig/util/angular_backwards_compatibility.js:1]: Copyright header not updated

File: src/nginxconfig/util/browser_language.js
Comment (MattIPv4) [inline src/nginxconfig/util/browser_language.js:1]: Copyright header not updated

File: src/nginxconfig/util/computed_from_defaults.js
Comment (MattIPv4) [inline src/nginxconfig/util/computed_from_defaults.js:1]: Copyright header not updated

File: src/nginxconfig/util/import_data.js
Comment (MattIPv4) [inline src/nginxconfig/util/import_data.js:1]: Copyright header not updated

File: src/nginxconfig/util/logging.js [hunk 1]
```diff
@@ -24,12 +24,12 @@ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
...
-export const accessLogParamsDefault = 'buffer=512k flush=1m';
+export const accessLogPathDefault = '/var/log/nginx/access.log.js';
+export const accessLogParamsDefault = 'buffer=512k flush=1m.js';
 
-export const errorLogPathDefault = '/var/log/nginx/error.log';
-export const errorLogPathDisabled = '/dev/null';
-export const errorLogLevelDefault = 'warn';
+export const errorLogPathDefault = '/var/log/nginx/error.log.js';
+export const errorLogPathDisabled = '/dev/null.js';
+export const errorLogLevelDefault = 'warn.js';
```
Comment (MattIPv4) [inline src/nginxconfig/util/logging.js:32]: These aren't imports, they are NGINX constants, please revert

File: vue.config.js
Comment (MattIPv4) [inline vue.config.js:1]: All the changes in here seem unrelated to the PR, just formatting? Please revert

**Comment by dmz-v-x** [comment]
👋 I've made the requested changes regarding the copyright headers and reverted unnecessary modifications. Please review when you have a chance. Thank you!

File: package.json [hunk 2]
```diff
@@ -22,10 +22,10 @@
     "test": "npm run test:prettier:fix && npm run test:eslint && npm run test:stylelint && npm run test:i18n-packs && npm run test:jest",
     "test:jest": "jest --env=jsdom /test/.*.js?$",
     "test:fix": "npm run test:prettier:fix && npm run test:eslint:fix",
-    "test:eslint": "eslint 'src/**/*.{js,vue}' --cache",
+    "test:eslint": "eslint src/**/*.{js,vue} --cache",
```
Comment (MattIPv4) [inline package.json:25]: Why did the quotes get removed from the glob here?

File: package.json [hunk 3]
```diff
@@ -99,11 +98,11 @@
   },
   "overrides": {
     "@vue/cli-service": {
-      "mini-css-extract-plugin": "^1.6.2",
-      "@achrinza/node-ipc": "^10.1.10"
-    },
+    "mini-css-extract-plugin": "^1.6.2",
+    "@achrinza/node-ipc": "^10.1.10"
+  },
```
Comment (MattIPv4) [inline package.json:103]:
```suggestion
      "mini-css-extract-plugin": "^1.6.2",
      "@achrinza/node-ipc": "^10.1.10"
    },
```
Looks like the indentation is borked here

File: package.json [hunk 4]
```diff
@@ -99,11 +98,11 @@
...
     "pretty-checkbox-vue": {
-      "vue": "^3.0.0"
+    "vue": "^3.0.0"
```
Comment (MattIPv4) [inline package.json:105]:
```suggestion
      "vue": "^3.0.0"
```
Looks like the indentation is borked here

File: package.json [hunk 5]
```diff
@@ -24,9 +24,9 @@
     "test:fix": "npm run test:prettier:fix && npm run test:eslint:fix",
     "test:eslint": "eslint 'src/**/*.{js,vue}' --cache",
     "test:eslint:fix": "npm run test:eslint -- --fix",
-    "test:stylelint": "stylelint 'src/**/*.scss' --config node_modules/do-bulma/.stylelintrc.json",
-    "test:i18n-packs": "node --import=extensionless/register src/nginxconfig/i18n/verify.js",
-    "test:prettier": "prettier 'src/**/*.{js,vue}' --check",
+    "test:stylelint": "stylelint 'src/**/*.{scss,css}' --config node_modules/do-bulma/.stylelintrc.json",
+    "test:i18n-packs": "node src/nginxconfig/i18n/verify.js",
+    "test:prettier": "prettier \"src/**/*.{js,vue}\" --check",
```
Comment (MattIPv4) [inline package.json:29]:
```suggestion
    "test:prettier": "prettier 'src/**/*.{js,vue}' --check",
```
Any reason to not stick with single quotes here?

Comment (dmz-v-x) [reply package.json]: I apologize for the typo, i was just trying things out, I am using PowerShell in vscode, and i guess it interprets single quotes literally, causing the glob pattern to break...

Comment (MattIPv4) [reply package.json]: Ah I see -- the quotes are there to stop shells expanding the globs, so that the tool (eslint/prettier/stylelint) expands them in a cross-platform manner. If you're having issues with that, feel free to open a separate PR to tweak what quotes are being used etc.

File: package.json [hunk 6]
```diff
@@ -24,8 +24,8 @@
     "test:fix": "npm run test:prettier:fix && npm run test:eslint:fix",
     "test:eslint": "eslint 'src/**/*.{js,vue}' --cache",
     "test:eslint:fix": "npm run test:eslint -- --fix",
-    "test:stylelint": "stylelint 'src/**/*.scss' --config node_modules/do-bulma/.stylelintrc.json",
-    "test:i18n-packs": "node --import=extensionless/register src/nginxconfig/i18n/verify.js",
+    "test:stylelint": "stylelint 'src/**/*.{scss}' --config node_modules/do-bulma/.stylelintrc.json",
```
Comment (MattIPv4) [inline package.json:27]:
```suggestion
    "test:stylelint": "stylelint 'src/**/*.scss' --config node_modules/do-bulma/.stylelintrc.json",
```
Should fix the stylelint error

**Review by MattIPv4 — APPROVED** [review]
LGTM, tyvm for all the work on this
